lib/mongoid/contextual/mongo.rb

```diff
@@ -420,20 +426,28 @@ def map_reduce(map, reduce)
       #
       # @return [ Array<Object, Array> ] The plucked values.
       def pluck(*fields)
+        # Multiple fields can map to the same field name. For example, plucking
+        # a field and its _translations field map to the same field in the database.
+        # because of this, we need to keep track of the fields inputted.
+        normalized_field_names = []
         normalized_select = fields.inject({}) do |hash, f|
-          hash[klass.database_field_name(f)] = 1
+          db_fn = klass.database_field_name(f)
+          normalized_field_names.push(db_fn)
+
+          if Mongoid.legacy_pluck_distinct
+            hash[db_fn] = 1
+          else
+            hash[klass.cleanse_localized_field_names(f)] = 1
+          end
           hash
         end
 
         view.projection(normalized_select).reduce([]) do |plucked, doc|
-          values = normalized_select.keys.map do |n|
-            res = n =~ /\./ ? doc[n.partition('.')[0]] : doc[n]
+          values = normalized_field_names.map do |n|
             if Mongoid.legacy_pluck_distinct
-              res
-            elsif field = klass.fields[n]
-              field.demongoize(res)
+              n =~ /\./ ? doc[n.partition('.')[0]] : doc[n]
```

Comment:
I see that the regexp match was already in the code, but I suggest `n.include?('.')` on the assumption that it is surely faster.

Reply:
done